esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5957,31 +5954,36 @@ void CWsDfuEx::getFileMeta(IPropertyTree &metaInfo, IDistributedFile &file, IUse
 #ifdef _USE_OPENSSL
     if (!isEmptyString(keyPairName)) // without it, meta data is not encrypted
     {
-        metaInfo.setProp("keyPairName", keyPairName);
-
-        Owned<IPropertyTree> secureMetaInfo = createPTreeFromIPT(&metaInfo);
-        secureMetaInfo->setProp("expiryTime", expiryTime);
-
-        extractFilePartInfo(*secureMetaInfo, *fDesc);
-
-        // 1st serialize to JSON
-        StringBuffer secureMetaInfoJson;
-        toJSON(secureMetaInfo, secureMetaInfoJson);
-
-        // 2nd compress
-        MemoryBuffer compressedSecureMetaInfoMb;
-        fastLZCompressToBuffer(compressedSecureMetaInfoMb, secureMetaInfoJson.length(), secureMetaInfoJson.str());
-
-        // 3rd encrypt
-        const char *publicKeyFName = env->getPublicKeyPath(keyPairName);
-        Owned<CLoadedKey> publicKey = loadPublicKeyFromFile(publicKeyFName, nullptr);
-        MemoryBuffer encryptedSecureMetaInfoMb;
-        aesEncryptWithRSAEncryptedKey(encryptedSecureMetaInfoMb, compressedSecureMetaInfoMb.length(), compressedSecureMetaInfoMb.bytes(), *publicKey);
-        metaInfo.setPropBin("secureInfo", encryptedSecureMetaInfoMb.length(), encryptedSecureMetaInfoMb.bytes());
+        metaInfo->setProp("version", securityInfoVersion);
+        metaInfo->setProp("logicalFilename", file.queryLogicalName());
+        metaInfo->setProp("jobId", req.getJobId());
+        metaInfo->setProp("accessType", req.getAccessTypeAsString());
+        StringBuffer userStr;
+        if (user)
+            metaInfo->setProp("user", user->getUserName(userStr).str());
+        metaInfo->setProp("keyPairName", keyPairName);
+        metaInfo->setProp("expiryTime", expiryTime);
+
+        MemoryBuffer metaInfoBlob;
+        metaInfo->serialize(metaInfoBlob);
+        metaInfoEnvelope->setPropBin("metaInfoBlob", metaInfoBlob.length(), metaInfoBlob.bytes());
+
+        const char *privateKeyFName = env->getPrivateKeyPath(keyPairName);
+        Owned<CLoadedKey> privateKey = loadPrivateKeyFromFile(privateKeyFName, nullptr);
+        StringBuffer metaInfoSignature;
+        digiSign(metaInfoSignature, metaInfoBlob.length(), metaInfoBlob.bytes(), *privateKey);
+        metaInfoEnvelope->setProp("signature", metaInfoSignature);
+        metaInfo->serialize(metaInfoMb);
+        metaInfoEnvelope->setPropBin("metaInfoBlob", metaInfoMb.length(), metaInfoMb.bytes());
```

Comment:
This appears to be a near duplicate of line 5968/5969

Reply:
you're right, but odd I'm surprised it didn't break something - probably because other end (in hpcc-20391) deserialized and ignored the duplicate trailing content at end.
Will fix.

Reply:
ah I see, not duplicated (in serialized buffer) but needs fixing